### PR TITLE
Detect and dump flaky LineMetrics test data

### PR DESCRIPTION
**Pull request by GaryQian**
PR meant to add additional logging to provide info on https://github.com/flutter/flutter/issues/43763 where a `LineMetrics` test is flaking on Cirrus.

### CODE REVIEW

**Comment by GaryQian** [comment]
Specifically, framework_tests-libraries-linux seems to be the task that flakes.

**Comment by GaryQian** [comment]
Ok, This has been modified to print data when the test would have failed. This will still continue to flake, but we will be able to see data this time.

**Comment by GaryQian** [comment]
Once this info dump is not necessary, this should be reverted.

**Review by Piinks — APPROVED** [review]
LGTM!

File: packages/flutter/test/painting/text_painter_test.dart [hunk 1]
```diff
@@ -745,6 +745,13 @@ void main() {
 
     expect(lines.length, 4);
 
+    if (lines[1].hardBreak == true) {
```
Comment (jonahwilliams) [inline packages/flutter/test/painting/text_painter_test.dart]: Can you leave a comment here with a link to the issue and a TODO to remove it?

Comment (GaryQian) [reply packages/flutter/test/painting/text_painter_test.dart]: Already did! just forgot to push it.

**Pull request merged**